src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -2311,7 +2366,7 @@ TEST_CASE("failed invocation with diagnostics", "[tx][soroban]")
     REQUIRE(contract_ev.event.type == ContractEventType::CONTRACT);
     REQUIRE(contract_ev.event.body.v0().data.type() == SCV_VEC);
 
-    auto const& read_entry_ev = opEvents.at(2);
+    auto const& read_entry_ev = opEvents.at(3);
```

Comment:
I'm not sure what's the event added after event [1]. I didn't want to spend much time debugging this, given that it's likely not a big issue.

Reply:
This is what I see (`VM call trapped: UnreachableCodeReached`)
```
{ inSuccessfulContractCall = FALSE,
    event = {
      ext = {
        v = 0
      },
      contractID = df06d62447fd25da07c0135eed7557e5a5497ee7d15b7fe345bd47e191d8f577,
      type = DIAGNOSTIC,
      body = {
        v = 0,
        v0 = {
          topics = [
            { type = SCV_SYMBOL,
              sym = "error" },
            { type = SCV_ERROR,
              error = {
                type = SCE_WASM_VM,
                code = SCEC_INVALID_ACTION
              } }
          ],
          data = {
            type = SCV_VEC,
            vec = [
              { type = SCV_STRING,
                str = "VM call trapped: UnreachableCodeReached" },
              { type = SCV_SYMBOL,
                sym = "add" },
              { type = SCV_ERROR,
                error = {
                  type = SCE_WASM_VM,
                  code = SCEC_INVALID_ACTION
                } }
            ]
          }
        }
      }
    } },
```

Reply:
Cool, this is actually expected - I did add additional logging for the VM traps.